Update token input box

its-miroma opened the pull request:
Fix part of #6000

## Screenshot

![Updated token section](https://github.com/refined-github/refined-github/assets/136986257/808da3e2-8887-4dda-a376-e1b6688572be)

its-miroma commented:
I'll lint it soon, sorry

fregante commented:
Thank you for the PR! This somewhat related to https://github.com/refined-github/refined-github/issues/7231, rather than #6000. 

I intentionally left the token visible to not lull the user into a false sense of security since the token is stored in clear text in Chrome's config. Also personally I leave it visible in order to reuse it across browsers/extensions  as needed. 

I'm not sure I want to make this specific change here, but I'm open for discussion

its-miroma commented:
> This somewhat related to https://github.com/refined-github/refined-github/issues/7231, rather than https://github.com/refined-github/refined-github/issues/6000.

I updated the description, to highlight the comment this PR is related to.

> I intentionally left the token visible to not lull the user into a false sense of security

I think it's wrong to think "Let's not implement X security feature because Y is not implemented", because they are two different problems: if you're recording/sharing the screen, you do not want to accidentally leak the token.

> the token is stored in clear text in Chrome's config

That's a different issue, that you can argue is way more critical, but it's not what I am trying to fix.

> I leave it visible in order to reuse it across browsers/extensions as needed

Of course, that's why I added a checkbox to display the token

fregante commented:
> you do not want to accidentally leak the token

This is mostly accomplished by the small field. The token is never shown in its entirety, but there are always 6+ characters missing

its-miroma commented:
Should I create a different PR to only add the username, as suggested in https://github.com/refined-github/refined-github/issues/6000#issuecomment-1556136109?

fregante commented:
Yes please that would be great
